Inherit from eth-utils CamelModel for pydantic types:

fselmo opened the pull request:
### What was wrong?

- The ``CustomPydanticModel`` used here inspired a more general class to be used across our libraries. That change was recently introduced in ``eth-utils``, so inherit from that class here for now, with a note to use ``CamelModel`` directly for the next major version.

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md)

#### Cute Animal Picture

<img width="474" alt="Screenshot 2025-04-14 at 16 15 17" src="https://github.com/user-attachments/assets/5390e0db-3a86-4ed0-8fe1-13eb555da09b" />

reedsa commented on setup.py:
```diff
@@ -55,7 +55,7 @@
         "eth-keys>=0.4.0",
         "eth-rlp>=2.1.0",
-        "eth-utils>=2.0.0",
+        "eth-utils>=5.3.0",
```
Does this merit a major or minor bump release? Curious what impact this may have.

kclowes replied:
Yeah, historically we've called a lower pin change like this a major. I think we should wait until our breaking cycle, but could be convinced otherwise :)

reedsa reviewed (approved):
lgtm, curious to hear thoughts about the next release.

fselmo committed:
Inherit from eth-utils CamelModel for pydantic types:

- The ``CustomPydanticModel`` used here inspired a more general
  class to be used across our libraries. That change was
  introduced in the latest ``eth-utils`` which we now use as a
  a bottom-pinned dependency.

fselmo committed:
remove deprecated method, no longer needed after eth-utils bump

fselmo committed:
parametrized sets -> ordered list so we can run with xdist

kclowes reviewed (approved):
🔥 much cleaner, nice! LGTM once tests pass!
